internal/wasm/store.go

```diff
@@ -116,6 +106,14 @@ type (
 		//	  security implications.
 		Sys *internalsys.Context
 
+		// Closed is used both to guard moduleEngine.CloseWithExitCode and to store the exit code.
+		//
+		// The update value is closedType + exitCode << 32. This ensures an exit code of zero isn't mistaken for never closed.
+		//
+		// Note: Exclusively reading and updating this with atomics guarantees cross-goroutine observations.
+		// See /RATIONALE.md
+		Closed atomic.Uint64
```

Comment:
same question about reordering. But maybe the note about keeping it at the top was only because it was a plain `uint64` to begin with?

Reply:
Yes. Heap pointers at the start of an allocation (start of heap allocated struct) are the only way to ensure 64-bit alignment before Go 1.19.